code/modules/mob/transform_procs.dm

```diff
@@ -3,16 +3,62 @@
 // A standardized proc for turning a mob into a monkey
 // ignore_primitive will force the mob to specifically become a monkey and not its primitive type
 // returns the monkey mob
-/mob/living/carbon/human/proc/monkeyize(var/ignore_primitive = 0)
+
+/mob/proc/Premorph(var/delete_items = FALSE)
 	if(monkeyizing)
-		return 0
-	monkeyizing = 1
+		return FALSE
+	regenerate_icons()
+	monkeyizing = TRUE
+	canmove = FALSE
+	icon = null
+	overlays.Cut()
+	invisibility = 101
+	delayNextAttack(50)
+
+	if(isrobot(src)) //Don't leave your brain behind.
+		var/mob/living/silicon/robot/sillycone = src
+		if(sillycone.mmi)
+			qdel(sillycone.mmi)
+	else
+		if(iscarbon(src))
```

Comment:
Wouldn't it be better if you did this in `mob/carbon/Premorph()` ? It would means less typechecks, I think.

Reply:
That's a pretty good idea